### PR TITLE
Do not run solver=classic tests in PRs

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Adds an environment variable to control the solvers exported by the fixture. It expects a comma separated list of strings (e.g. `libmamba,classic`).

<details>

<summary>Timings in main</summary>

Integration tests for Windows & Linux take almost 1h for each job if the default channel is conda-forge.

| Windows Integration 1 | `main` |
|-----------------------|--------|
| tests/cli/test_env.py::test_conda_env_create_http[classic] | 526.82s call |
| tests/cli/test_main_remove.py::test_remove_globbed_package_names[classic] | 282.98s call |
| tests/test_create.py::test_dont_remove_conda_2[classic] | 257.13s call |
| tests/cli/test_env.py::test_create_valid_env_with_conda_and_pip_json_output[classic] | 208.85s call |
| tests/test_create.py::test_conda_pip_interop_pip_clobbers_conda[classic] | 122.29s call |
| tests/test_create.py::test_channel_usage_replacing_python[libmamba] | 84.23s call |
| tests/test_create.py::test_strict_resolve_get_reduced_index[classic] | 73.46s call |
| tests/test_create.py::test_shortcut_creation_installs_shortcut[classic] | 53.32s call |
| tests/test_create.py::test_update_all_updates_pip_pkg[libmamba] | 52.05s call |
| tests/test_create.py::test_noarch_python_package_reinstall_on_pyver_change[libmamba] | 47.09s call |
| tests/test_create.py::test_conda_pip_interop_conda_editable_package[classic] | 45.55s call |
| tests/test_create.py::test_noarch_python_package_with_entry_points[libmamba] | 44.36s call |
| tests/core/test_solve.py::test_freeze_deps_1[classic] | 43.48s call |
| tests/env/test_create.py::test_create_env_default_packages | 43.13s call |
| tests/test_create.py::test_update_all_updates_pip_pkg[classic] | 43.08s call |
| tests/cli/test_env.py::test_update_env_only_pip_json_output[libmamba] | 38.71s call |

| Windows Integration 2 | `main` |
|-----------------------|--------|
| tests/cli/test_env.py::test_update_env_only_pip_json_output[classic] | 402.99s call |
| tests/test_create.py::test_conda_downgrade[classic] | 324.85s call |
| tests/cli/test_cli_install.py::test_find_conflicts_called_once[classic] | 290.01s call |
| tests/test_create.py::test_conda_downgrade[libmamba] | 256.51s call |
| tests/test_create.py::test_force_remove[classic] | 180.75s call |
| tests/test_create.py::test_create_install_update_remove_smoketest[classic] | 154.47s call |
| tests/test_create.py::test_create_install_update_remove_smoketest[libmamba] | 116.99s call |
| tests/test_create.py::test_strict_resolve_get_reduced_index[libmamba] | 73.87s call |
| tests/test_create.py::test_dont_remove_conda_1[libmamba] | 72.42s call |
| tests/test_create.py::test_dont_remove_conda_2[libmamba] | 64.08s call |
| tests/test_create.py::test_list_with_pip_wheel[libmamba] | 54.81s call |
| tests/test_create.py::test_conda_pip_interop_compatible_release_operator[libmamba] | 51.08s call |
| tests/test_create.py::test_shortcut_absent_does_not_barf_on_uninstall[classic] | 50.53s call |
| tests/test_create.py::test_shortcut_creation_installs_shortcut[libmamba] | 49.65s call |
| tests/test_create.py::test_shortcut_absent_when_condarc_set[libmamba] | 49.14s call |
| tests/test_create.py::test_noarch_python_package_with_entry_points[classic] | 48.91s call |

| Windows Integration 3 | `main` |
|-----------------------|--------|
| tests/cli/test_env.py::test_update_env_no_action_json_output[classic] | 532.92s call |
| tests/test_create.py::test_dont_remove_conda_1[classic] | 272.38s call |
| tests/cli/test_env.py::test_pip_error_is_propagated[classic] | 204.77s call |
| tests/test_create.py::test_conda_pip_interop_compatible_release_operator[classic] | 170.34s call |
| tests/test_create.py::test_channel_usage_replacing_python[classic] | 155.08s call |
| tests/test_create.py::test_cross_channel_incompatibility[classic] | 121.94s call |
| tests/test_create.py::test_conda_pip_interop_conda_editable_package[libmamba] | 76.51s call |
| tests/test_create.py::test_install_prune_flag[classic] | 74.91s call |
| tests/test_create.py::test_json_create_install_update_remove[libmamba] | 69.15s call |
| tests/test_create.py::test_shortcut_absent_does_not_barf_on_uninstall[libmamba] | 61.30s call |
| tests/test_create.py::test_list_with_pip_wheel[classic] | 57.60s call |
| tests/test_create.py::test_noarch_python_package_without_entry_points[classic] | 56.26s call |
| tests/test_create.py::test_conda_pip_interop_pip_clobbers_conda[libmamba] | 55.12s call |
| tests/test_create.py::test_shortcut_absent_when_condarc_set[classic] | 47.62s call |
| tests/cli/test_env.py::test_update_env_no_action_json_output[libmamba] | 42.27s call |
| tests/test_create.py::test_safety_checks_disabled[libmamba] | 39.32s call |

| Linux Integration 1 | `main` |
|-----------------------|--------|
| tests/cli/test_main_remove.py::test_remove_globbed_package_names[classic] | 501.59s call |
| tests/test_create.py::test_conda_pip_interop_compatible_release_operator[classic] | 335.41s call |
| tests/test_create.py::test_create_install_update_remove_smoketest[classic] | 294.98s call |
| tests/cli/test_env.py::test_update_env_json_output[classic] | 225.76s call |
| tests/test_create.py::test_force_remove[classic] | 219.67s call |
| tests/test_create.py::test_create_install_update_remove_smoketest[libmamba] | 104.12s call |
| tests/test_create.py::test_strict_resolve_get_reduced_index[libmamba] | 66.92s call |
| tests/test_create.py::test_strict_resolve_get_reduced_index[classic] | 64.97s call |
| tests/test_create.py::test_json_create_install_update_remove[libmamba] | 51.21s call |
| tests/test_create.py::test_dont_remove_conda_1[libmamba] | 41.80s call |
| tests/test_create.py::test_disallowed_packages[classic] | 37.11s call |
| tests/test_create.py::test_conda_pip_interop_conda_editable_package[classic] | 34.83s call |
| tests/test_create.py::test_conda_pip_interop_compatible_release_operator[libmamba] | 31.83s call |
| tests/cli/test_env.py::test_export_multi_channel[libmamba] | 29.88s call |
| tests/core/test_solve.py::test_freeze_deps_1[classic] | 28.15s call |
| tests/test_activate.py::test_stacking[5-base,has-has-has,base,sys] | 26.34s setup |

| Linux Integration 2 | `main` |
|-----------------------|--------|
| tests/cli/test_cli_install.py::test_find_conflicts_called_once[classic] | 546.48s call |
| tests/test_create.py::test_dont_remove_conda_2[classic] | 363.28s call |
| tests/test_create.py::test_conda_pip_interop_pip_clobbers_conda[classic] | 308.35s call |
| tests/test_create.py::test_conda_downgrade[libmamba] | 257.59s call |
| tests/test_create.py::test_cross_channel_incompatibility[classic] | 199.67s call |
| tests/test_create.py::test_noarch_python_package_with_entry_points[classic] | 108.58s call |
| tests/cli/test_env.py::test_create_valid_env_with_conda_and_pip_json_output[classic] | 106.27s call |
| tests/cli/test_env.py::test_pip_error_is_propagated[classic] | 105.55s call |
| tests/cli/test_env.py::test_update[classic] | 105.24s call |
| tests/cli/test_env.py::test_env_export_with_variables[classic] | 77.86s call |
| tests/test_create.py::test_dont_remove_conda_2[libmamba] | 38.24s call |
| tests/test_create.py::test_offline_with_empty_index_cache[classic] | 32.07s call |
| tests/test_create.py::test_conda_pip_interop_pip_clobbers_conda[libmamba] | 31.29s call |
| tests/test_activate.py::test_stacking[5-base,not-has-has,base,sys] | 28.80s setup |
| tests/test_create.py::test_list_with_pip_wheel[libmamba] | 28.04s call |
| tests/test_activate.py::test_stacking[5-base,has-base-base,has,base,sys] | 27.58s setup |

| Linux Integration 3 | `main` |
|-----------------------|--------|
| tests/cli/test_env.py::test_conda_env_create_http[classic] | 516.02s call |
| tests/test_create.py::test_dont_remove_conda_1[classic] | 356.09s call |
| tests/test_create.py::test_conda_downgrade[classic] | 273.08s call |
| tests/cli/test_env.py::test_update_env_only_pip_json_output[classic] | 209.76s call |
| tests/test_create.py::test_json_create_install_update_remove[classic] | 199.16s call |
| tests/cli/test_env.py::test_update_env_no_action_json_output[classic] | 191.51s call |
| tests/test_create.py::test_neutering_of_historic_specs[classic] | 155.14s call |
| tests/test_create.py::test_install_prune_flag[classic] | 114.34s call |
| tests/test_create.py::test_noarch_python_package_without_entry_points[classic] | 103.31s call |
| tests/cli/test_env.py::test_export_multi_channel[classic] | 102.21s call |
| tests/cli/test_env.py::test_env_export[classic] | 100.97s call |
| tests/test_create.py::test_channel_usage_replacing_python[classic] | 63.19s call |
| tests/test_create.py::test_channel_usage_replacing_python[libmamba] | 50.30s call |
| tests/test_create.py::test_list_with_pip_wheel[classic] | 37.16s call |
| tests/test_create.py::test_conda_pip_interop_conda_editable_package[libmamba] | 36.64s call |
| tests/test_create.py::test_neutering_of_historic_specs[libmamba] | 36.56s call |

</details>

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
Using miniforge as the base installation didn't change much. Still ~1h per job.

**Comment by jezdez** [comment]
Since we know that the classic solver isn't responding well to large repodata, which conda-forge is, I think we should try to dynamically disable the classic solver based tests if the default channel is conda-forge. It's unreasonable to use CI resources for a use case we know is broken, basically.

**Comment by jaimergp** [comment]
In `main`:


| Windows       | main | no classic |
|---------------|------|------------|
| Integration 1 | 54m  | 31m38s     |
| Integration 2 | 60m  | 32m38s     |
| Integration 3 | 60m  | 29m45s     |

| Linux         | main | no classic |
|---------------|------|------------|
| Integration 1 | 49m  | 17m30s     |
| Integration 2 | 52m  | 16m30s     |
| Integration 3 | 58m  | 21m40s     |

**Comment by jezdez** [comment]
> In `main`:
> 
> Windows	main	no classic
> Integration 1	54m	31m38s
> Integration 2	60m	32m38s
> Integration 3	60m	29m45s
> Linux	main	no classic
> Integration 1	49m	17m30s
> Integration 2	52m	16m30s
> Integration 3	58m	21m40s

Thank's for conducting these trials, I think this is significant CI performance improvement to warrant moving the classic solver tests into a scheduled run (once per day?) instead of keeping them in the regular iteration.

**Comment by jaimergp** [comment]
pre-commit.ci fix

**Comment by jaimergp** [comment]
This PR:

image


Main:

image

**Comment by jezdez** [comment]
Shall we add another periodic run _with_ classic, in main?

**Comment by jaimergp** [comment]
This kind of line:

```
CONDA_TEST_SOLVERS: ${{ github.event_name == 'pull_request' && 'libmamba' || 'libmamba,classic' }}
```

Makes sure we do test classic (along with libmamba) on every push to main and tag release. Do you want to add a cronjob too? I think this repo receives enough activity a week to not require cronjobs.

**Review by jezdez — APPROVED** [review]

File: .github/TEST_FAILURE_REPORT_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+title: "{{ env.TITLE }} ({{ date | date("YYYY-MM-DD") }})"
+labels: [type::bug, type::testing]
```
Comment (travishathaway) [inline .github/TEST_FAILURE_REPORT_TEMPLATE.md:3]: Could we add `source::auto` so we can easily filter issues that come from automation?

**Review by travishathaway — CHANGES_REQUESTED** [review]
One requested change for the automatic issue creation. We have `source::auto` so let's use it 💪 .

File: .github/TEST_FAILURE_REPORT_TEMPLATE.md (hunk 1 above)
Comment (jaimergp) [reply .github/TEST_FAILURE_REPORT_TEMPLATE.md:3]: Oh nice touch!

Comment (jaimergp) [reply .github/TEST_FAILURE_REPORT_TEMPLATE.md:3]:
```suggestion
labels: [type::bug, type::testing, source::auto]
```

**Review by travishathaway — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**